system/src/Grav/Console/Gpm/DirectInstallCommand.php

```diff
@@ -0,0 +1,381 @@
+
+                } else {
+                    $this->output->write("\x0D");
+                    $this->output->writeln("  |- Checking destination...  <green>ok</green>");
+                }
+
+                $this->output->write("  |- Installing package...  ");
+                Installer::install($zip, GRAV_ROOT, ['sophisticated' => true, 'overwrite' => true, 'ignore_symlinks' => true], $extracted);
+            } else {
+                $name = $this->getPackageName($extracted);
+
+                if (!$name) {
+                    $this->output->writeln("<red>ERROR: Name could not be determined.</red> Please specify with --name|-n");
+                    $this->output->writeln('');
+                    exit;
+                }
+
+                $install_path = $this->getInstallPath($type, $name);
+                $is_update = file_exists($install_path);
+
+                $this->output->write("  |- Checking destination...  ");
+
+                Installer::isValidDestination(GRAV_ROOT . DS . $install_path);
+                if (Installer::lastErrorCode() == Installer::IS_LINK) {
+                    $this->output->write("\x0D");
+                    $this->output->writeln("  |- Checking destination...  <yellow>symbolic link</yellow>");
+                    $this->output->writeln("  '- <red>ERROR: symlink found...</red>  <yellow>" . GRAV_ROOT . DS . $install_path . '</yellow>');
+                    $this->output->writeln('');
+                    exit;
+
+                } else {
+                    $this->output->write("\x0D");
+                    $this->output->writeln("  |- Checking destination...  <green>ok</green>");
+                }
+
+                $this->output->write("  |- Installing package...  ");
+
+                Installer::install($zip, GRAV_ROOT, ['install_path' => $install_path, 'theme' => (($type == 'theme')), 'is_update' => $is_update], $extracted);
+            }
+
+            Folder::delete($tmp_source);
+
+            $this->output->write("\x0D");
+
+            if(Installer::lastErrorCode()) {
+                $this->output->writeln("  '- <red>Installation failed or aborted.</red>");
+                $this->output->writeln('');
+            } else {
+                $this->output->writeln("  |- Installing package...    <green>ok</green>");
+                $this->output->writeln("  '- <green>Success!</green>  ");
+                $this->output->writeln('');
+            }
+
+        } else {
+            $this->output->writeln("  '- <red>ERROR: ZIP package could not be found</red>");
+        }
+
+        Folder::delete($tmp_zip);
+
+        // clear cache after successful upgrade
+        $this->clearCache();
+
+        return true;
+
+    }
+
+    /**
+     * Determine if Grav is symlinked (system is all we need to check)
+     *
+     * @return bool
+     */
+    protected function gravSymlinked()
+    {
+        if (is_link(GRAV_ROOT . DS . 'system')) {
+            return true;
+        }
+    }
```

Comment:
This is not needed, `Installer` as an internal method to verify validity of destination:

```php
Installer::isValidDestination(GRAV_ROOT . '/system');
if (Installer::IS_LINK === Installer::lastErrorCode()) { return false; } // Grav is symlinked
```